src/App.js

```diff
@@ -2,12 +2,66 @@ import React from 'react';
 
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './components/GoodsList/GoodsList';
 // or
 // import * as goodsAPI from './api/goods';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+class App extends React.Component {
+  state = {
+    goods: [],
+    isListVisible: false,
+  }
+
+  showAll = () => {
+    getAll()
+      .then((goods) => {
+        this.setState(prevState => ({
+          goods,
+          isListVisible: !prevState.isListVisible,
```

Comment:
You don't need isListVisible property. 
You need to always render this.state.goods, and change it's value by callbacks. 
When I press showAll button, all the goods are rendered; show5First - first 5 are rendered. 
Now it doesn't work the way it should